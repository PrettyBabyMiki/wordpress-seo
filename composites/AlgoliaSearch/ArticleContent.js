import React from "react";

/**
 * Creates the JSX to render the content of the selected article.
 *
 * @param {object} props The React props.
 * @returns {ReactElement} A div with the content of the selected article.
 * @constructor
 */
const ArticleContent = ( props ) => {
	let url = props.post.permalink + "amp?source=wpseo-kb-search";

	return ( <iframe src={ url } className="kb-search-content-frame" title={ props.iframeTitle } /> );
};

ArticleContent.propTypes = {
	post: React.PropTypes.object,
	iframeTitle: React.PropTypes.string,
};

ArticleContent.defaultProps = {
	iframeTitle: "Knowledge base article",
};

export default ArticleContent;
