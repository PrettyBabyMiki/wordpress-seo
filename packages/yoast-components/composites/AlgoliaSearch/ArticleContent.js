import React from "react";
import IFrame from "../basic/IFrame";
import PropTypes from "prop-types";
import styled from "styled-components";

const ArticleContentFullWidth = styled( IFrame )`
	width: 100%;
	height: 600px;
	border: none;
`;

/**
 * Creates the JSX to render the content of the selected article.
 *
 * @param {object} props The React props.
 * @returns {ReactElement} An iFrame with the content of the selected article.
 * @constructor
 */

/**
 * The ArticleContent component.
 */
class ArticleContent extends React.Component {
	/**
	 * Constructs the article content.
	 *
	 * @param {Object} props The props for the article content.
	 */
	constructor( props ) {
		super( props );
	}

	/**
	 * Transforms a given URL to an AMP URL.
	 *
	 * @param {string} originalUrl The original URL.
	 *
	 * @returns {string} The AMP URL.
	 */
	toAmp( originalUrl ) {
		return `${ originalUrl }amp/?source=wpseo-kb-search`;
	}

	/**
	 * Renders the article content with the AMP URL.
	 *
	 * @returns {ReactElement} The rendered element.
	 */
	render() {
		return ( <ArticleContentFullWidth
			src={ this.toAmp( this.props.post.permalink ) }
			className="kb-search-content-frame"
			title={ this.props.title }
		/> );
	}
}

ArticleContent.propTypes = {
	post: PropTypes.object.isRequired,
	title: PropTypes.string,
};

ArticleContent.defaultProps = {
	title: "Knowledge base article",
};

export default ArticleContent;
