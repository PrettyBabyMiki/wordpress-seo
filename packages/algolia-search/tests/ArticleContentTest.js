import React from "react";
import renderer from "react-test-renderer";

import ArticleContent from "../src/ArticleContent";

test( "the ArticleContent component  matches the snapshot", () => {
	const component = renderer.create(
		<ArticleContent post={ { permalink: "www.example.com/" } } title="KB article" />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
