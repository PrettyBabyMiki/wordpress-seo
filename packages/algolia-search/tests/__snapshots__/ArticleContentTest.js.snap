// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ArticleContent component  matches the snapshot 1`] = `
.c0 {
  width: 100%;
  height: 600px;
  border: none;
}

<iframe
  className="kb-search-content-frame c0"
  src="www.example.com/amp/?source=wpseo-kb-search"
  title="KB article"
/>
`;
